PlaylistManager/Downloaders/PlaylistSequentialDownloader.cs

```diff
@@ -266,11 +266,10 @@ private async Task BeatSaverBeatmapDownload(Beatmap song, BeatmapVersion songver
                 Directory.CreateDirectory(customSongsPath);
             }
 
-            if (!ownedHashes.Contains(songversion.Hash.ToUpper()))
+            if (ownedHashes.Add(songversion.Hash))
```

Comment:
Thinking about it, this might not be correct since it would add the song to the collection even if the zip extraction fails.... Thoughts?